TestServerDoesNotLeakGoroutines is flakey

<!--
/area autoscale
/kind cleanup
/assign josephburnett
-->

## Expected Behavior

statserver:TestServerDoesNotLeakGoroutines should pass consistently.

## Actual Behavior

but it's flakey

## Steps to Reproduce the Problem

```
$ go test ./pkg/autoscaler/statserver/... -count 1000

--- FAIL: TestServerDoesNotLeakGoroutines (5.57s)
        server_test.go:137: Current number of goroutines 5 is not equal to the original number 6
--- FAIL: TestServerDoesNotLeakGoroutines (5.50s)
        server_test.go:137: Current number of goroutines 6 is not equal to the original number 7
--- FAIL: TestServerDoesNotLeakGoroutines (5.51s)
        server_test.go:137: Current number of goroutines 7 is not equal to the original number 8
FAIL
FAIL    github.com/knative/serving/pkg/autoscaler/statserver    32.215s
```

--- a/pkg/autoscaler/statserver/server_test.go
+++ b/pkg/autoscaler/statserver/server_test.go
@@ -129,7 +129,7 @@ func TestServerDoesNotLeakGoroutines(t *testing.T) {
 	// Check the number of goroutines eventually reduces to the number there were before the connection was created
 	for i := 1000; i >= 0; i-- {
 		currentGoRoutines := runtime.NumGoroutine()
-		if currentGoRoutines == originalGoroutines {
+		if currentGoRoutines <= originalGoroutines {
 			break
 		}
 		time.Sleep(5 * time.Millisecond)

